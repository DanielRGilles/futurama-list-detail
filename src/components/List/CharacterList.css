.container {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr 1fr ;
    grid-auto-flow: row;
    grid-gap: 5px;
    max-width: 100%;
    margin: 0;
    justify-items: center;
    align-content: center;
}
@media only screen and (max-width: 1700px) {
    .container {
      grid-template-columns: 1fr 1fr 1fr 1fr;
    }
  }
@media only screen and (max-width: 1400px) {
    .container {
      grid-template-columns: 1fr 1fr 1fr;
    }
  }
  @media only screen and (max-width: 1000px) {
    .container {
      grid-template-columns: 1fr 1fr;
    }
  }
  @media only screen and (max-width: 600px) {
    .container {
      grid-template-columns: 1fr;
      max-width: 90%;
    }
    .character > img {
        height: 50px;
    }
  }
.character {
    display: flex;
    flex-direction: column;
    border: 2px solid black;
    margin: 5px;
    padding: 15px;
    max-width: 75%;
    border-radius: 5px;
    background-color: cadetblue;
    font-size: 75%;
    
}
.character > img {
    width: auto;
    height: fit-content;
    min-height: 50px;
    max-height: 300px;

}
.character > div {
    margin-top: 5%;
}

.grass {
    background-color: lightgreen;
  }
  .fire {
    background-color: lightsalmon;
  }
  .water {
    background-color: lightblue;
  }
  .bug {
    background-color: rgb(247, 247, 37);
  }
  .normal {
    background-color: lightgrey;
  }
  .steel {
    background-color: slategray;
  }
  .psychic {
    background-color: rgb(124, 83, 165);
  }
  .dragon {
    background-color: rgba(19, 16, 16, 0.705);
    color: coral;
  }
  .dark {
    background-color: gold;
  }
  .ground {
    background-color: brown;
  }
  .ice {
    background-color: #D6ECEF
  }
  .fighting {
    background-color: wheat;
  }
  .fairy {
    background-color: pink;
  }
  .ghost {
    background-color: lightslategray;
  }
  .flying {
    background-color: coral;
  }
  .poison {
    background-color: darkgreen;
  }
  .rock {
    background-color: rgb(87, 87, 26);
  }
  .electric {
    background-color: skyblue;
    font-size: .8rem;
  }

  