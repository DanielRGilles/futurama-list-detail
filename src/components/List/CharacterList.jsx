import React from 'react'
import styles from './CharacterList.css'
import { Link } from 'react-router-dom'
import '../../App.css'

export default function CharacterList({ characters }) {
    return (
        <div className={styles.container}>
           {characters.map(({id, image, name, type}) => <div  className={styles[`${type}`] + ' ' + styles.character} key={id}>
                <img src={image} alt={name} />
               <Link to={`/characters/${id}`}><h1>{ name }</h1></Link>
               <div>Type: {type}</div>
              
           </div>)} 
           {console.log([characters])}
        </div>
    )
}
