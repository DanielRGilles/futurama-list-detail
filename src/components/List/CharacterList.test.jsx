import {render } from  '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import CharacterList from './CharacterList';



describe('Header component test', () => {
  
    it('renders the top nav header', () => {
       const { container } = render(
            <MemoryRouter>
                
                     <CharacterList characters={characters}/>
               
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
         
    })

})



const characters = [
    {
        "id": "5ff4fb7cd89993a89cc6563f",
        "name": "abomasnow",
        "image": "http://assets.pokemon.com/assets/cms2/img/pokedex/full/460.png",
        "type": "grass",
        "color": "#7FCB70"
    },
    {
        "id": "5ff4fb7cd89993a89cc65641",
        "name": "abomasnow-mega",
        "image": "http://assets.pokemon.com/assets/cms2/img/pokedex/full/460_f2.png",
        "type": "grass",
        "color": "#7FCB70"
    },
    {
        "id": "5ff4fb7cd89993a89cc65488",
        "name": "abra",
        "image": "http://assets.pokemon.com/assets/cms2/img/pokedex/full/063.png",
        "type": "psychic",
        "color": "NA"
    },
    {
        "id": "5ff4fb7cd89993a89cc655cb",
        "name": "absol",
        "image": "http://assets.pokemon.com/assets/cms2/img/pokedex/full/359.png",
        "type": "dark",
        "color": "NA"
    },
]
