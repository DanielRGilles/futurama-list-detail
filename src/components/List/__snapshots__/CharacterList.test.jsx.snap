// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test renders the top nav header 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="grass character"
    >
      <img
        alt="abomasnow"
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/460.png"
      />
      <a
        href="/characters/5ff4fb7cd89993a89cc6563f"
      >
        <h1>
          abomasnow
        </h1>
      </a>
      <div>
        Type: 
        grass
      </div>
    </div>
    <div
      class="grass character"
    >
      <img
        alt="abomasnow-mega"
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/460_f2.png"
      />
      <a
        href="/characters/5ff4fb7cd89993a89cc65641"
      >
        <h1>
          abomasnow-mega
        </h1>
      </a>
      <div>
        Type: 
        grass
      </div>
    </div>
    <div
      class="psychic character"
    >
      <img
        alt="abra"
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/063.png"
      />
      <a
        href="/characters/5ff4fb7cd89993a89cc65488"
      >
        <h1>
          abra
        </h1>
      </a>
      <div>
        Type: 
        psychic
      </div>
    </div>
    <div
      class="dark character"
    >
      <img
        alt="absol"
        src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/359.png"
      />
      <a
        href="/characters/5ff4fb7cd89993a89cc655cb"
      >
        <h1>
          absol
        </h1>
      </a>
      <div>
        Type: 
        dark
      </div>
    </div>
  </div>
</div>
`;
